pinot-spi/src/main/java/org/apache/pinot/spi/data/Schema.java

```diff
@@ -499,6 +499,15 @@ public SchemaBuilder addSingleValueDimension(String dimensionName, DataType data
       return this;
     }
 
+    /**
+     * Add single value dimensionFieldSpec with maxLength and a defaultNullValue
+     */
+    public SchemaBuilder addSingleValueDimension(String dimensionName, DataType dataType, int maxLength,
+                                                 Object defaultNullValue) {
```

Comment:
Please reformat this

Reply:
Gotcha, thanks @Jackie-Jiang for code review.